### PR TITLE
chore: configure gitpod

### PR DESCRIPTION
**Pull request by vil02**
This PR allows to use this repository in gitpod.

In the created environment, it is possible to successfully run the commands

```shell
busted -o utfTerminal .spec
```

and

```shell
luacheck .
```

Besides the fact that several repositories in [`TheAlgorithms`](https://github.com/TheAlgorithms) are _ready to use_ in gitpod, I think it is a good idea to have it, since it shows the newcomers how to create their own development environment.

### CODE REVIEW

**Review by Panquesito7 — APPROVED** [review]
Looks great. Thanks! 🚀

File: .gitpod.dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM gitpod/workspace-base:2023-11-04-12-07-48
+
+USER root
+
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends luarocks=3.8.0+dfsg1-1 \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
```
Comment (appgurueu) [inline .gitpod.dockerfile:8]: This line removes the package lists. Why?

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+image:
+  file: .gitpod.dockerfile
+
+tasks:
+  - init: |
+      echo "Welcome to TheAlgorithms/Lua"
```
Comment (appgurueu) [inline .gitpod.yml:7]: Why is this useful? It seems to just lead to confusing spam in the console:

Screenshot from 2023-11-08 13-33-26

**Review by appgurueu — COMMENTED** [review]
Frankly, I'm not much of a fan of Gitpod; I don't really see the benefits.

All this does for a newcomer is set up a crippled Ubuntu image with Lua, Luarocks, Luacheck and Busted installed in a "cloud development environment". To actually commit their changes, users would further need to grant wide-reaching permissions to Gitpod. I don't think this is how newcomers should approach this, depending on Gitpod just to save them 5 minutes of setting up the dependencies.

Ultimately, programmers - and this includes newcomers to this repo - need to be able to set up a dev environment. Given the few and simple "dependencies" of this repo, I see no need to involve Docker - over a 3rd-party service, no less - yet; again, I don't think we should lead newcomers to use Gitpod.

(And even if Gitpod were to lower the barrier to entry, that won't necessarily be a good thing; I doubt that someone who would have been unable to set up a dev environment is able to make a substantial contribution.)

---

That said, to "show the newcomers how to create their own development environment", we should probably add a "getting started" section to the Readme - adding some links to the Lua / Luarocks / Luacheck / Busted installation / getting started pages, perhaps even a few ready-to-go commands and instructions for the most popular operating systems for convenience.

Newcomers *will* have to deal with dev env setup either way; neither Luacheck nor Busted invoke themselves, and for Luacheck I highly recommend editor support; it doesn't even seem possible to install a Luacheck extension on Gitpod out of the box:


Screenshot from local VSC:
Screenshot from 2023-11-08 13-43-40
Screenshot from Gitpod:
Screenshot from 2023-11-08 13-43-08

(perhaps this is fixable?)

---

If Gitpod support were accepted, this PR would also be missing a mention of Gitpod in the Readme.

File: .gitpod.dockerfile (hunk 1 above)
Comment (vil02) [reply .gitpod.dockerfile:8]: It it reduces the image size, cf. [DL3009](https://github.com/hadolint/hadolint/wiki/DL3009).

File: .gitpod.yml (hunk 2 above)
Comment (vil02) [reply .gitpod.yml]: Removed in 1d2e149e0c79b2847f0058f465e58eb9a7562b5e.

**Comment by vil02** [comment]
@appgurueu I think that the _cost_ of maintaining gitpod setup is minimal. The benefits are that (gitpod users) can contribute from a browser. If I would not find it useful, I would not create this PR.

File: .gitpod.dockerfile (hunk 1 above)
Comment (appgurueu) [reply .gitpod.dockerfile:8]: I'm not sure that the advice applies here. For a deployment setup (what docker is usually used for), you don't need the ability to install new packages, but a development setup shouldn't be "bricked" - devs may want to install further packages.

**Review by appgurueu — APPROVED** [review]
I'm not strongly opposed to this; it is true that the maintenance cost is limited.

File: .gitpod.dockerfile (hunk 1 above)
Comment (vil02) [reply .gitpod.dockerfile:8]: They can still run `sudo apt-get update`.

**Comment by appgurueu** [comment]
Thanks for the PR!

**Pull request merged**